portability-core/src/test/java/org/dataportabilityproject/serviceProviders/google/contacts/GoogleContactsServiceTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2018 The Data-Portability Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dataportabilityproject.serviceProviders.google.contacts;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.api.services.people.v1.PeopleService;
+import com.google.api.services.people.v1.PeopleService.People.Connections;
+import com.google.api.services.people.v1.model.EmailAddress;
+import com.google.api.services.people.v1.model.FieldMetadata;
+import com.google.api.services.people.v1.model.ListConnectionsResponse;
+import com.google.api.services.people.v1.model.Name;
+import com.google.api.services.people.v1.model.Person;
+import ezvcard.VCard;
+import ezvcard.property.Email;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import org.dataportabilityproject.cloud.interfaces.JobDataCache;
+import org.dataportabilityproject.cloud.local.InMemoryJobDataCache;
+import org.dataportabilityproject.dataModels.ExportInformation;
+import org.dataportabilityproject.dataModels.contacts.ContactsModelWrapper;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GoogleContactsServiceTest {
+
+  private final Logger logger = LoggerFactory.getLogger(GoogleContactsServiceTest.class);
+
+  private static final String FIRST_NAME = "Jane";
+  private static final String LAST_NAME = "Doe";
+  private static final String RESOURCE_NAME = "resource_name";
+  private static final FieldMetadata PRIMARY_FIELD_METADATA = new FieldMetadata().setPrimary(true);
+  private static final Name NAME = new Name()
+      .setGivenName(FIRST_NAME)
+      .setFamilyName(LAST_NAME)
+      .setMetadata(PRIMARY_FIELD_METADATA);
+  private static final Person PERSON = new Person()
+      .setNames(Collections.singletonList(NAME))
+      .setResourceName(RESOURCE_NAME);
+
+  private Connections.List connectionsListRequest = mock(Connections.List.class);
```

Comment:
You might want to initialize these in a method annotated with @Before that way you can a fresh mock object and cache for each test and avert issues arising from state across tests as we add more

Reply:
Excellent idea!